### ISSUE TITLE
tight_layout doesn't leave enough space between outwards ticks and axes title

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#5868]
All's in the title (notice how the descender of the "j" (also pretty bad with "g") runs into the outwards tick).
```
from pylab import *
rcParams["xtick.direction"] = "out"
gca().set_title("j" * 30)
tight_layout()
show()
```

1.5.0
![screenshot_20160117_020324](https://cloud.githubusercontent.com/assets/1322974/12376904/d605844c-bcbe-11e5-9398-257010f4fc6a.png)
master
![screenshot_20160117_020348](https://cloud.githubusercontent.com/assets/1322974/12376905/d60ad816-bcbe-11e5-8d3c-b86c142a522f.png)

**Comment by tacaswell** [comment]
@mdboom Do you have a fix for this on an un-merged branch?

**Comment by mdboom** [comment]
#5683 (merged to 2.x) is somewhat related, but only includes the outward ticks in the tight layout calculation -- it doesn't address the tick-to-text relationship.

We don't really have any automatic tick-to-text work at present.  The planned fix in #5774 is just to increase the distance between the spine and the tick labels in the defaults -- and styles with inward ticks would set the space to some lower value.

However, I wouldn't be opposed to automatically adding the tick length to the spine-to-tick label distance when the ticks are outward if we can find a clean way to do it that doesn't have negative consequences.

All this, like many other things, is really crying out for a proper constraint-based layout engine that we don't have.  (And I often wonder how much brittle and hard-to-maintain ad hoc layout code would go away if we had a constraint engine)...  but I digress...

**Comment by tacaswell** [comment]
https://github.com/nucleic/kiwi is said engine (and it is already shipped
by conda).  The hard part is writing all of the rules...

**Comment by efiring** [comment]
Fixed by #6468
